frontend/src/redux/Login/types.ts

```diff
@@ -0,0 +1,16 @@
+export const LOGIN = 'LOGIN';
+
+export interface User {
+  userId: number;
+  username: string;
+}
+
+export interface LoginAction {
+  type: typeof LOGIN;
```

Comment:
Something fishy about this const string being used with `typeof` here and in a `switch` statement in the reducer. Right now `typeof LOGIN` is just `string` right?

Reply:
Indeed. Action types aren't necessarily strings, which is why I didn't hard code it here. I took this from the Redux doc